### ISSUE TITLE
multiprocessing.pool not compatible with -OO flag and PyInstaller

### ISSUE THREAD

**Issue by b2m** [uqfoundation/pathos#159]
**Environment**

Pathos Version: 0.2.2.1
Python Version: 3.6.6

**Problem description**

I am trying to generate an [optimized python](https://docs.python.org/3.6/using/cmdline.html#cmdoption-oo) executable using [PyInstaller](https://www.pyinstaller.org/). As with the optimizing the docstrings get removed the following line from [multiprocessing.py](https://github.com/uqfoundation/pathos/blob/943c809f637e144c9f34f24ce7b5b90a29368ba2/pathos/multiprocessing.py#L111) produces an error when starting the executable.

**The "offending" code**
```python
__init__.__doc__ = AbstractWorkerPool.__init__.__doc__ + __init__.__doc__
```

**The error message**
```
TypeError: unsupported operand type(s) for +: 'NoneType' and 'NoneType'
```

**For reproduction**
Here is a script to reproduce the error using a docker image.
```bash
# on your command lie
docker run -it cdrx/pyinstaller-linux:python3 /bin/bash
```

```bash
# in the docker container
pip install pathos
echo "from pathos.multiprocessing import Pool
if __name__ == "__main__":
     with Pool() as pool:
        pool.map(print, range(10))" > testscript.py
python3 -OO -m PyInstaller  --onefile --noconfirm --clean --name test-pathos testscript.py
./dist/test-pathos
```

**Workaround**

Use only `-O` for optimization with the consequence of having all the docstrings in your executable.

**Comment by mmckerns** [comment]
I think I've addressed your issue in abfebcd8f282ad1d8e38d2fa0623bb7819c0c75d.  If that doesn't work, please reopen the ticket and comment.  Thanks for the detailed bug report.

**Comment by b2m** [comment]
Just confirming that the reported bug is solved on the current master branch.
